### ISSUE TITLE
Speech to Text for audio files

### ISSUE THREAD

**Issue by AdhocAdam** [AdhocAdam/smletsexchangeconnector#54]
If an organization redirects their phone system's voicemail to send emails to the Service Manager workflow account inbox one can use the Speech API from Azure Cognitive services to convert the audio file to text and insert it into the Work Item description, Action Log, or the Description of the respective wav file.

This also entertains the possibility of issuing voice commands to SCSM/System Center at large.

**Comment by AdhocAdam** [comment]
Through Azure Cognitive Services it's possible to perform speech analysis through the Speech API. Take the following PowerShell function with a deployed Azure Cognitive Service Speech API. Passing in your own api key and wav file on disk you can translate audio to text.

```powershell
$azureCogSvcSpeechAPIKey = ""

function Get-AzureEmailAudioText ($waveFileToEvaluate)
{
    $SpeechServiceURI = "https://speech.platform.bing.com/speech/recognition/dictation/cognitiveservices/v1?language=en-us&format=simple"

    # $OAuthToken is the authorization token returned by the token service.
    $RecoRequestHeader = @{
        'Ocp-Apim-Subscription-Key' = "$azureCogSvcSpeechAPIKey";
        'Transfer-Encoding' = 'chunked'
        'Content-type' = 'audio/wav; codec=audio/pcm; samplerate=16000'
    }

    # Read audio into byte array
    $audioBytes = [System.IO.File]::ReadAllBytes($waveFileToEvaluate)

    $RecoResponse = Invoke-RestMethod -Method POST -Uri $SpeechServiceURI -Headers $RecoRequestHeader -Body $audioBytes

    # Show the result
    $RecoResponse
}

Get-AzureEmailAudioText -waveFileToEvaluate "test.wav"
```

**Comment by AdhocAdam** [comment]
I'd be interested to know if people have custom message classes for their voicemails or they appear as simply native IPM.Note (i.e. an email).

**Comment by AdhocAdam** [comment]
With [Bing Speech being retired and getting replaced with Speech Services](https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/how-to-migrate-from-bing-speech), this requires a change in how the PowerShell call is made. Fortunately, it isn't all that different and still returns the identical object as before making migration a rather painless experience. This newer service requires not just the API key but a slightly more specific URL based on which region you've deployed Speech Services into. Take the following example using the Azure East US region:

```powershell
$azureCogSvcSpeechAPIKey = ""

function Get-AzureSpeechEmailAudioText ($waveFileToEvaluate)
{  
    #build the request
    $SpeechServiceURI = "https://eastus.stt.speech.microsoft.com/speech/recognition/conversation/cognitiveservices/v1?language=en-us"
    $RecoRequestHeader = @{
      'Ocp-Apim-Subscription-Key' = "$azureCogSvcSpeechAPIKey";
      'Content-type' = "audio/wav; codecs=audio/pcm; samplerate=16000";
      'Transfer-Encoding' = 'chunked'
      'Except' = "100-continue"
      'Accept' = "application/json"
    }

    # Read audio into byte array and submit the request
    $audioBytes = [System.IO.File]::ReadAllBytes($waveFileToEvaluate)
    $RecoResponse = Invoke-RestMethod -Method POST -Uri $SpeechServiceURI -Headers $RecoRequestHeader -Body $audioBytes

    # Show the result
    $RecoResponse
}

Get-AzureSpeechEmailAudioText -waveFileToEvaluate "test.wav"
```

Further information such as customizing the header and region specific URLs for the new Speech Services REST API can be found [here](https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/rest-text-to-speech).

**Comment by AdhocAdam** [comment]
Will be released with #197 and #198
